core/modules/image/image.field.inc

```diff
@@ -162,13 +164,36 @@ function image_field_instance_settings_form($field, $instance) {
     '#description' => t('The alt attribute may be used by search engines, screen readers, and when the image cannot be loaded.'),
     '#weight' => 10,
   );
+  $form['alt_field_required'] = array(
+    '#type' => 'checkbox',
+    '#title' => t('<em>Alt</em> field required'),
+    '#default_value' => $settings['alt_field_required'],
+    '#description' => t('Requiring alternative text is strongly encouraged to improve accessibility for all users.'),
+    '#weight' => $form['alt_field']['#weight'] + .1,
```

Comment:
Weight is usually an integer. Is this intentional or an oversight?

```suggestion
    '#weight' => $form['alt_field']['#weight'] + 1,
```

Reply:
I believe it's to make it more likely that the checkbox will be directly below the alt field option rather than drop below other settings fields. Though I'd need to confirm. It's the same here in `user_form_field_ui_field_edit_form_alter()`. Maybe it's not needed if it's not altering an existing form.